### PR TITLE
Budget Model

### PR DESCRIPTION
**Pull request by lipwei1808**
Creates Budget Model
- Budget class
- Interval class
- BudgetInterval enum (day, week, month)

Refactored Amount class to unicash.model.commons

### CODE REVIEW

File: src/main/java/unicash/commons/enums/TransactionType.java [hunk 1]
```diff
@@ -34,18 +34,18 @@ public static TransactionType parseType(String transactionType) {
     }
 
     /**
-     * Returns a boolean value if a given string is a valid TransactionType
+     * Returns a boolean value if a given string is a valid TransactionType.
      *
-     * @param test a string value to be tested
-     * @return a boolean value if the given string is a valid enum
+     * @param test a string value to be tested.
+     * @return a boolean value if the given string is a valid enum.
```
Comment (sp4ce-cowboy) [inline src/main/java/unicash/commons/enums/TransactionType.java:40]: nice attention to detail

File: src/main/java/unicash/model/transaction/Transaction.java [hunk 2]
```diff
@@ -6,9 +6,10 @@
 
 import unicash.commons.util.ToStringBuilder;
 import unicash.model.category.UniqueCategoryList;
+import unicash.model.commons.Amount;
 
 /**
- * Represents a Transaction in the finance tracker.
+ * Represents a Transaction in the UniCash.
```
Comment (sp4ce-cowboy) [inline src/main/java/unicash/model/transaction/Transaction.java:12]:
```suggestion
 * Represents a Transaction in UniCash.
```

File: src/main/java/unicash/model/budget/Budget.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+import static unicash.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+
+import unicash.commons.util.ToStringBuilder;
+import unicash.model.commons.Amount;
+
+/**
+ * Represents a Budget in the UniCash.
```
Comment (sp4ce-cowboy) [inline src/main/java/unicash/model/budget/Budget.java:11]:
```suggestion
 * Represents a Budget in UniCash.
```

File: src/main/java/unicash/model/budget/Interval.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
+package unicash.model.budget;
+
+import static java.util.Objects.requireNonNull;
+import static unicash.commons.util.AppUtil.checkArgument;
+
+import unicash.commons.enums.BudgetInterval;
+
+/**
+ * Represents a Transaction's interval.
```
Comment (sp4ce-cowboy) [inline src/main/java/unicash/model/budget/Interval.java:9]: I'm not exactly sure what this means, does it mean like the Interval between two transactions? i.e. one day/week/month between each transaction?

**Review by sp4ce-cowboy — APPROVED** [review]
LGTM, just some minor clarifications and suggestions. Although since there are quite a bit of additions, might wanna wait for someone else to cross review

File: src/main/java/unicash/model/budget/Interval.java (hunk 4 above)
Comment (lipwei1808) [reply src/main/java/unicash/model/budget/Interval.java:9]: Actually a typo. Supposed to be Budget interval

**Review by sp4ce-cowboy — APPROVED** [review]
LGTM

**Pull request merged**